Initial streamz docker image to lower barrier for users accessing str…

jdye64 opened the pull request:
…eamz exmaples. Closes: #287

CJ-Wright commented on requirements.txt:
```diff
@@ -2,3 +2,5 @@ tornado
 toolz
 zict
 six
+numpy
```
These are optional deps I think so we might consider if they are needed or not.

jdye64 replied:
That is a good point. Let me remove them from requirements.txt and rather just install them with pip directly in the Dockerfile itself. 

If the conda PR merges at another point that could also be used instead to make the Dockerfile even cleaner.

jdye64 commented:
@CJ-Wright I modified the Dockerfile to use the conda environment from the optional dependencies now instead of pip. This also ensures that as things are added to the conda environment file over time they will be automatically added to the Docker container at build time.

I also added some documentation on building in docs/index.rst. Let me know what you think.

martindurant commented:
@CJ-Wright , were you reviewing this?

martindurant commented:
ping @CJ-Wright

CJ-Wright commented:
Sorry for the delay

CJ-Wright commented on Dockerfile:
```diff
@@ -0,0 +1,47 @@
+
+# Install system dependencies and convenience utilities
+RUN apt-get update -y && \
+    apt-get install -y wget \
```
Is there any reason not to get these two from conda-forge?

CJ-Wright commented:
This looks good to me, one minor comment.

jdye64 commented:
Sorry for the major delay folks. Yep agree its dumb of my to not just add them via conda. I'll make that change and commit now after a validation build.

CJ-Wright commented:
cycle for ci

CJ-Wright commented:
This is failing on flake8 I'll push a fix to the branch

CJ-Wright commented:
I don't know why flake8 is unhappy some of the offending lines don't even exist at the most recent commit.

martindurant commented:
I see no reason not the merge this - @CJ-Wright , agreed?
